api/v1alpha1/clusterconfig_types.go

```diff
@@ -237,84 +241,113 @@ func (ExtraAPIServerCertSANs) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
-type ImageRegistries struct {
+type ImageCredentials struct {
+	// The Secret containing the registry credentials and optional CA certificate
+	// using the keys `username`, `password` and `ca.crt`.
+	// This credentials Secret is not required for some registries, e.g. ECR.
 	// +optional
-	ImageRegistryCredentials ImageRegistryCredentials `json:"credentials,omitempty"`
+	SecretRef *corev1.ObjectReference `json:"secretRef,omitempty"`
 }
 
-func (ImageRegistries) VariableSchema() clusterv1.VariableSchema {
+func (ImageCredentials) VariableSchema() clusterv1.VariableSchema {
 	return clusterv1.VariableSchema{
 		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
-			Description: "Configuration for image registries.",
-			Type:        "object",
+			Type: "object",
 			Properties: map[string]clusterv1.JSONSchemaProps{
-				"credentials": ImageRegistryCredentials{}.VariableSchema().OpenAPIV3Schema,
+				"secretRef": {
+					Description: "The Secret containing the registry credentials. " +
+						"The Secret should have keys 'username', 'password' and optional 'ca.crt' " +
+						"This credentials Secret is not required for some registries, e.g. ECR.",
+					Type: "object",
+					Properties: map[string]clusterv1.JSONSchemaProps{
+						"name": {
+							Description: "The name of the Secret containing the registry credentials.",
+							Type:        "string",
+						},
+						"namespace": {
+							Description: "The namespace of the Secret containing the registry credentials. " +
+								"Defaults to the namespace of the KubeadmControlPlaneTemplate and KubeadmConfigTemplate " +
+								"that reference this variable.",
```

Comment:
This should default to the namespace of the `Cluster` resource as that is where this variable is used.
```suggestion
								"Defaults to the namespace of the Cluster.",
```